playbook/initializeDomain.ts

```diff
@@ -20,35 +25,94 @@ task('initializeDomain', 'Initialize domain name and activate it')
         discount,
         registrarAddress,
         activate,
-      }: { domain: string; freeRegistrationsNumber: string; fee: string; reward: string; discount: string; registrarAddress: string; activate: boolean },
+        username,
+      }: { domain: string; freeRegistrationsNumber: string; fee: string; reward: string; discount: string; registrarAddress: string; activate: boolean; username: string },
       hre,
     ) => {
       const { deployments, getNamedAccounts } = hre;
-      const { owner, registrar } = await getNamedAccounts();
+      const { owner, registrar, player } = await getNamedAccounts();
       const multipassDeployment = await deployments.get('Multipass');
       registrarAddress = registrarAddress ?? registrar;
       const multipassContract = new hre.ethers.Contract(
         multipassDeployment.address,
         multipassDeployment.abi,
         hre.ethers.provider.getSigner(owner),
       ) as MultipassDiamond;
-      const tx = await multipassContract.initializeDomain(
+      /*const tx = await multipassContract.initializeDomain(
         registrarAddress,
         freeRegistrationsNumber,
         hre.ethers.utils.parseEther(fee),
         hre.ethers.utils.formatBytes32String(domain),
         hre.ethers.utils.parseEther(reward),
         hre.ethers.utils.parseEther(discount),
       );
-      console.log(tx.wait(1));
+      console.log(tx.wait(1));*/
 
       if(activate === true) {
-        const tx = await multipassContract
+        /*const tx = await multipassContract
           .activateDomain(hre.ethers.utils.formatBytes32String(domain));
         console.log(tx.wait(1));
-        console.log('Domain name "' + domain + '" successfully initialized and activated!')
+        console.log('Domain name "' + domain + '" successfully initialized and activated!')*/
+      }
+
+      if(username) {
+        const playerId = crypto.randomUUID().slice(0, 31);
+        const registrarMessage = {
+          name: hre.ethers.utils.formatBytes32String(username),
+          id: hre.ethers.utils.formatBytes32String(playerId),
+          domainName: hre.ethers.utils.formatBytes32String(domain),
+          deadline: hre.ethers.BigNumber.from(9999),
+          nonce: hre.ethers.BigNumber.from(0),
+        };
+        let signer =  await hre.ethers.getSigner(player);
+    
+        const validSignature = await signRegistrarMessage(registrarMessage, multipassDeployment.address, signer, hre);
```

Comment:
signer on `signRegistrarMessage` must be `registrar` of the `Multipass` contract.

Reply:
It seems you are trying to sign with `player` instead.